patches/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -132,36 +140,48 @@
          float f = this.getHealth();
          if (f > 0.0F) {
              this.setHealth(f + p_21116_);
-@@ -1118,6 +_,7 @@
- 
-     @Override
-     public boolean hurt(DamageSource p_21016_, float p_21017_) {
-+        if (!net.neoforged.neoforge.common.CommonHooks.onLivingAttack(this, p_21016_, p_21017_)) return false;
-         if (this.isInvulnerableTo(p_21016_)) {
+@@ -1127,23 +_,26 @@
+         } else if (p_21016_.is(DamageTypeTags.IS_FIRE) && this.hasEffect(MobEffects.FIRE_RESISTANCE)) {
              return false;
-         } else if (this.level().isClientSide) {
-@@ -1136,14 +_,17 @@
+         } else {
++            this.damageContainer = this.damageContainer == null ? new net.neoforged.neoforge.common.damagesource.DamageContainer.InternalDamageContainer(p_21016_, p_21017_) : this.damageContainer;
++            if (!net.neoforged.neoforge.common.CommonHooks.onEntityPreDamage(this, this.damageContainer)) return false;
+             if (this.isSleeping() && !this.level().isClientSide) {
+                 this.stopSleeping();
+             }
+ 
+             this.noActionTime = 0;
+-            float f = p_21017_;
++            float f = p_21017_ = this.damageContainer.getNewDamage();
```

Comment:
This patch can be changed to the addition of `p_21017_ = this.damageContainer.getNewDamage();` above the `float f = p_21017_;` line instead of patching it out.  It should also include a `// Neo: ` comment stating why we override the parameter instead of just the local.